### PR TITLE
Fix timestamps to milliseconds

### PR DESCRIPTION
**Pull request by reedstrm**
This is why everything was January 19th, 1970. :)

### CODE REVIEW

File: app/controllers/api/v0/events_controller.rb [hunk 1]
```diff
@@ -77,11 +77,12 @@ def fixup_timestamps
...
       occurred_at = delete(:client_clock_occurred_at)
 
-      self[:occurred_at] = TimeUtil.infer_actual_occurred_at_from_client_timestamps(
+      # Schema expects integer timestamps in milliseconds
+      self[:occurred_at] = (TimeUtil.infer_actual_occurred_at_from_client_timestamps(
         request_received_at: controller.request.env[:received_at],
         client_clock_occurred_at: occurred_at,
         client_clock_sent_at: sent_at
-      ).to_i
+      ).to_f * 1000).to_i
```
Comment (jpslav) [inline app/controllers/api/v0/events_controller.rb]: wonder if we'd want to monkey patch `Time` to be explicit here with

```ruby
class Time
  def floored_milliseconds_since_epoch
    (to_f * 1000).floor
  end
end
```

(`to_i` is doing a floor)

and then make this code and in the spec be

```ruby
TimeUtil.infer_actual_occurred_at_from_client_timestamps(
  ...
).floored_milliseconds_since_epoch
```

would let us get rid of the `#Schema expects integer timestamps in milliseconds` comment

Not critical, just a thought.  Your call.

Comment (reedstrm) [reply app/controllers/api/v0/events_controller.rb]: Yeah, I also like the code to explain itself. Forgot that extending a class like this is that easy. Will do.

Comment (reedstrm) [reply app/controllers/api/v0/events_controller.rb]: Ok extended, added a spec, now everything works _except_ the request specs. Method words from rails console. Added extension to config/initializers. Soooo, why are the 6 POST specs failing?
example:
```
.....................F.FFFFF.....................

Failures:

  1) Api::V0::EventsController POST /events successfully calls the API and sends a data message to kafka
     Failure/Error:
       self[:occurred_at] = TimeUtil.infer_actual_occurred_at_from_client_timestamps(
         request_received_at: controller.request.env[:received_at],
         client_clock_occurred_at: occurred_at,
         client_clock_sent_at: sent_at
       ).floored_milliseconds_since_epoch
     
     NoMethodError:
       undefined method `floored_milliseconds_since_epoch' for Fri, 26 Aug 73211 06:47:12 +0000:DateTime
     # ./app/controllers/api/v0/events_controller.rb:84:in `fixup_timestamps'
     # ./app/controllers/api/v0/events_controller.rb:38:in `initialize'
     # ./app/controllers/api/v0/events_controller.rb:9:in `new'
     # ./app/controllers/api/v0/events_controller.rb:9:in `block in create'
     # ./app/controllers/api/v0/events_controller.rb:8:in `each'
     # ./app/controllers/api/v0/events_controller.rb:8:in `create'
     # ./ruby/2.6.0/gems/sentry-raven-3.0.0/lib/raven/integrations/rails/controller_transaction.rb:7:in `block in included'
     # ./ruby/2.6.0/gems/scout_apm-3.0.0.pre28/lib/scout_apm/middleware.rb:20:in `call'
     # ./ruby/2.6.0/gems/versionist-2.0.1/lib/versionist/middleware.rb:39:in `_call'
     # ./ruby/2.6.0/gems/versionist-2.0.1/lib/versionist/middleware.rb:17:in `call'
     # ./ruby/2.6.0/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
     # ./ruby/2.6.0/gems/rack-2.2.3/lib/rack/conditional_get.rb:40:in `call'
     # ./ruby/2.6.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
     # ./ruby/2.6.0/gems/lograge-0.11.2/lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
     # ./ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/rack/logger.rb:26:in `block in call'
     # ./ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/rack/logger.rb:26:in `call'
     # ./ruby/2.6.0/gems/openstax_healthcheck-0.0.3/lib/openstax/healthcheck/middleware.rb:13:in `call'
     # ./ruby/2.6.0/gems/request_store-1.5.0/lib/request_store/middleware.rb:19:in `call'
     # ./ruby/2.6.0/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
     # ./ruby/2.6.0/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
     # ./app/middlewares/request_received_at.rb:8:in `call'
     # ./ruby/2.6.0/gems/sentry-raven-3.0.0/lib/raven/integrations/rack.rb:51:in `call'
     # ./ruby/2.6.0/gems/rack-cors-1.1.1/lib/rack/cors.rb:100:in `call'
     # ./ruby/2.6.0/gems/railties-5.2.4.3/lib/rails/engine.rb:524:in `call'
     # ./ruby/2.6.0/gems/rack-test-1.1.0/lib/rack/mock_session.rb:29:in `request'
     # ./ruby/2.6.0/gems/rack-test-1.1.0/lib/rack/test.rb:266:in `process_request'
     # ./ruby/2.6.0/gems/rack-test-1.1.0/lib/rack/test.rb:119:in `request'
     # ./spec/requests/api/v0/events_request_spec.rb:153:in `block (3 levels) in <top (required)>'
     # ./ruby/2.6.0/gems/webmock-3.9.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

[...]
Failed examples:

rspec ./spec/requests/api/v0/events_request_spec.rb:148 # Api::V0::EventsController POST /events successfully calls the API and sends a data message to kafka
```

Comment (reedstrm) [reply app/controllers/api/v0/events_controller.rb]: done as requested.

**Review by jpslav — APPROVED** [review]

**Pull request merged**